packages/debug/src/browser/editor/debug-editor-service.ts

```diff
@@ -84,38 +84,47 @@ export class DebugEditorService {
         return uri && this.models.get(uri.toString());
     }
 
-    get logpoint(): DebugBreakpoint | undefined {
-        const logpoint = this.anyBreakpoint;
+    getContextLogpoint(position?: monaco.Position): DebugBreakpoint | undefined {
+        if (!position) {
+            return undefined;
+        }
+        const logpoint = this.anyBreakpoint(position);
         return logpoint && logpoint.logMessage ? logpoint : undefined;
     }
-    get logpointEnabled(): boolean | undefined {
-        const { logpoint } = this;
+    getContextLogpointEnabled(position?: monaco.Position): boolean | undefined {
+        const logpoint = this.getContextLogpoint(position);
         return logpoint && logpoint.enabled;
     }
 
-    get breakpoint(): DebugBreakpoint | undefined {
-        const breakpoint = this.anyBreakpoint;
+    getContextBreakpoint(position?: monaco.Position): DebugBreakpoint | undefined {
```

Comment:
I don't think we need `Context` marker. This method can be used programmatically to get a breakpoint at some position from other context.

Also if does not do anything for `undefined` value it is strange to have a signature allowing to call this method without any argument. Would not it better to move check that a position is defined to context editor command handlers? i.e. `isEnabled: position => monaco.Position.is(position) && !!this.getBreakpoint(position)`

The same for others methods.

Reply:
The `context` name was specifically because it was only used in the context menus. I can rename.

> Would not it better to move check that a position is defined to context editor command handlers and make the position argument mandatory?

Personally I think this is a lot harder to read, there are many functions which return if the optional arguments are undefined.

However, happy to make changes to get this PR merged ASAP

Reply:
ok, let's go with it, I hope nobody relies on removed APIs

Reply:
Unfortunately there is no `monaco.Position.is(position)` and it's in typefox code :(